lms/static/js/spec/ccx/schedule_spec.js

```diff
@@ -0,0 +1,197 @@
+                val = "i4x://edX/DemoX/sequential/edx_introduction";
+                view.sequential_select.val(val);
+                view.sequential_select.change();
+                expect(view.sequential_select.val()).toEqual(val);
+                expect(view.vertical_select.prop('disabled')).toEqual(false);
+                expect(view.vertical_select.children('option').length).toEqual(2);
+            });
+
+            it("selects a section", function() {
+                var val = 'i4x://edX/DemoX/chapter/d8a6192ade314473a78242dfeedfbf5b';
+                view.chapter_select.val(val);
+                view.chapter_select.change();
+                val = "i4x://edX/DemoX/sequential/edx_introduction";
+                view.sequential_select.val(val);
+                view.sequential_select.change();
+                val = "i4x://edX/DemoX/vertical/vertical_0270f6de40fc",
+                view.vertical_select.val(val);
+                view.vertical_select.change();
+                expect(view.vertical_select.val()).toEqual(val);
+            });
+
+            it("adds a unit", function() {
+                var val = 'i4x://edX/DemoX/chapter/d8a6192ade314473a78242dfeedfbf5b';
+                view.chapter_select.val(val);
+                view.chapter_select.change();
+                val = "i4x://edX/DemoX/sequential/edx_introduction";
+                view.sequential_select.val(val);
+                view.sequential_select.change();
+                val = "i4x://edX/DemoX/vertical/vertical_0270f6de40fc",
+                view.vertical_select.val(val);
+                view.vertical_select.change();
+                var unit = view.find_unit(view.schedule, 'i4x://edX/DemoX/chapter/d8a6192ade314473a78242dfeedfbf5b');
+                expect(unit.hidden).toBe(true);
+                $('#add-unit-button').click();
+                expect(unit.hidden).toBe(false);
+            });
+
+            it("gets a datetime string from date and time fields", function() {
+                view.set_datetime('start', '2015-12-12 10:45');
+                expect($('form#add-unit input[name=start_date]')).toHaveValue('2015-12-12');
+                expect($('form#add-unit input[name=start_time]')).toHaveValue('10:45');
+            });
+
+            it("sets date and time fields from datetime string", function() {
+                $('form#add-unit input[name=start_date]').val('2015-12-12');
+                $('form#add-unit input[name=start_time]').val('10:45');
+                var datetime = view.get_datetime('start');
+                expect(datetime).toBe('2015-12-12 10:45');
+            });
+
+            it("saves schedule changes", function() {
+                view.save();
+                assertAjax("save_ccx", "POST", data);
+                expect($('#ajax-error')).toHaveCss({display: 'none'});
+            });
+
+            it("displays an error if the sync fails", function() {
+                ajaxSuccess = false;
```

Comment:
This is *really* confusing -- it took me awhile to identify how setting this variable was related to this test. Another reason why you should use Sinon to mock out `XMLHttpRequest`, rather than doing your own custom mocking.